pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -1103,11 +1103,42 @@ func (b *MasqueradeBindMechanism) createNatRulesUsingNftables(proto iptables.Pro
 		return err
 	}
 
+	addressesToDnat, err := b.getDstAddressesToDnat(proto)
+	if err != nil {
+		return err
+	}
+
 	if len(b.iface.Ports) == 0 {
-		err = b.handler.NftablesAppendRule(proto, "nat", "KUBEVIRT_PREINBOUND",
+		if !hasIstioSidecarInjectionEnabled(b.vmi) {
+			err = b.handler.NftablesAppendRule(proto, "nat", "KUBEVIRT_PREINBOUND",
+				"counter", "dnat", "to", b.getDhcpConfigIpByProtocol(proto))
+			if err != nil {
+				return err
+			}
```

Comment:
Don't we have `return nil` since this is all we need to do if there is no istio sidecar

Reply:
Originally we just do the following, so I suppose we have to do the same if there is no sidecar injection enabled
```
if len(b.iface.Ports) == 0 {                                                
        err = b.handler.NftablesAppendRule(proto, "nat", "KUBEVIRT_PREINBOUND", 
            "counter", "dnat", "to", b.getDhcpConfigIpByProtocol(proto))        
                                                                                
        return err                                                              
    }                                                                           
```

Reply:
https://github.com/kubevirt/kubevirt/pull/5461#discussion_r634952505